test/integration/acceptance/grantable_permissions_fixture.hpp

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_GRANTABLE_PERMISSIONS_FIXTURE_HPP
+#define IROHA_GRANTABLE_PERMISSIONS_FIXTURE_HPP
+
+#include <gtest/gtest.h>
+
+#include "builders/protobuf/queries.hpp"
+#include "builders/protobuf/transaction.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+
+class GrantablePermissionsFixture : public AcceptanceFixture {
+ public:
+  using TxBuilder = TestUnsignedTransactionBuilder;
+
+  /**
+   * Create a transaction that creates a user and a role with specified name and
+   * a set of permissions. The default domain role will be detached from the
+   * user and the new role will be attached to the user.
+   * @param user - user name without domain
+   * @param key - keypair used to sign user creation tx
+   * @param perms - permissions to be contained in role
+   * @param role - role name to be attached to the user
+   * @return proto::Transaction
+   */
+  shared_model::proto::Transaction makeAccountWithPerms(
+      const shared_model::interface::types::AccountNameType &user,
+      const shared_model::crypto::Keypair &key,
+      const shared_model::interface::RolePermissionSet &perms,
+      const shared_model::interface::types::RoleIdType &role);
+
+  /**
+   * Creates two accounts with corresponding permission sets
+   * @param itf - initialized instance of test framework
+   * @param perm1 set of permissions for account #1
+   * @param perm2 set of permissions for account #2
+   * @return reference to ITF object with two transactions
+   */
+  integration_framework::IntegrationTestFramework &createTwoAccounts(
+      integration_framework::IntegrationTestFramework &itf,
+      const shared_model::interface::RolePermissionSet &perm1,
+      const shared_model::interface::RolePermissionSet &perm2);
+
+  /**
+   * Create a transaction such that the creator grants permittee a permission
+   * @param creator_account_name - first's account name without domain
+   * @param creator_key - the keypair to sign the transaction
+   * @param permitee_account_name - a name of permittee account (receives the
+   * grantable permission)
+   * @param grant_permission - grantable permisssion to be granted
+   * @return proto::Transaction
+   */
+  shared_model::proto::Transaction accountGrantToAccount(
```

Comment:
Maybe just name it grantPermission? It is clear from parameter names that it is from account to account